src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -434,6 +434,18 @@ internal bool ReportErrors(IEnumerable<Diagnostic> diagnostics, TextWriter conso
         private bool ReportErrors(DiagnosticBag diagnostics, TextWriter consoleOutput, ErrorLogger errorLoggerOpt)
             => ReportErrors(diagnostics.ToReadOnly(), consoleOutput, errorLoggerOpt);
 
+        internal static bool HasUnsuppressedErrors(DiagnosticBag diagnostics)
```

Comment:
The name threw me off here a bit. Felt like has not fake error. Consider a more positive term like HasReal/ReportedError.

Reply:
Suppressions may or may not be reported. A warning elevated to an error isn't a "real" error to me. This is the most accurate term I can think of.